File: JS-Basics/06-Conditional-Statements-Advanced-Exercise/p03_New_House.js
function newHouse(input) {
    //----Input-------------
    let typeFlower = input[0];
    let countFlowers = Number(input[1]);
    let budget = Number(input[2]);

    //-----table----price---per---one--flower----
    let rosePrice = 5;
    let dahliaPrice = 3.80;
    let tulipsPrice = 2.80;
    let narcissusPrice = 3;
    let gladiolusPrice = 2.50;


    let currentPrice = 0;
    let totalPrice = 0;
    let discount = 0;
    //------TODO---------logic-------
    switch (typeFlower) {
        case 'Roses':
            currentPrice = countFlowers * rosePrice;
            if (countFlowers > 80) {
                discount = currentPrice * 0.1;
            }
            totalPrice = currentPrice - discount;
            break;
        case 'Dahlias':
            currentPrice = countFlowers * dahliaPrice;
            if (countFlowers > 90) {
                discount = currentPrice * 0.15;
            }
            totalPrice = currentPrice - discount;
            break;
        case 'Tulips':
            currentPrice = countFlowers * tulipsPrice;
            if (countFlowers > 80) {
                discount = currentPrice * 0.15;
            }
            totalPrice = currentPrice - discount;
            break;
        case 'Narcissus':
            currentPrice = countFlowers * narcissusPrice;
            if (countFlowers < 120) {
                discount = currentPrice * 0.15;
            }
            totalPrice = currentPrice + discount;
            break;
        case 'Gladiolus':
            currentPrice = countFlowers * gladiolusPrice;
            if (countFlowers < 80) {
                discount = currentPrice * 0.20;
            }
            totalPrice = currentPrice + discount;
            break;
    }

    // ----output-----------
    if (budget >= totalPrice) {
        console.log(`Hey, you have a great garden with ${countFlowers} ${typeFlower} and ${(budget - totalPrice).toFixed(2)} leva left.`)
    } else {
        console.log(`Not enough money, you need ${(totalPrice - budget).toFixed(2)} leva more.`)
    }
}

newHouse(["Roses", "55", "250"]);
newHouse(["Tulips", "88", "260"]);

File: Programming-Basic/06-Conditional-Statements-Advanced-Exercise/p03_New_House.java
import java.util.Scanner;

public class p03_New_House {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

//        От конзолата се четат 3 реда:
//•	Вид цветя - текст с възможности - "Roses", "Dahlias", "Tulips", "Narcissus", "Gladiolus"//•	Брой цветя - цяло число в интервала [10…1000]
//•	Бюджет - цяло число в интервала [50…2500]
        String flowers = scanner.nextLine();
        int numFlowers = Integer.parseInt(scanner.nextLine());
        int budget = Integer.parseInt(scanner.nextLine());

        double expenses = 0.0;

        // proverka po vid cvetq
        switch (flowers) {
            case "Roses":
                expenses = numFlowers * 5.0;

                if (numFlowers > 80) {
                    expenses *= 0.90;
                }

                break;
            case "Dahlias":
                expenses = numFlowers * 3.80;

                if (numFlowers > 90) {
                    expenses *= 0.85;
                }

                break;
            case "Tulips":
                expenses = numFlowers * 2.80;

                if (numFlowers > 80) {
                    expenses *= 0.85;
                }

                break;
            case "Narcissus":
                expenses = numFlowers * 3.0;

                if (numFlowers < 120) {
                    expenses *= 1.15;
                }

                break;
            case "Gladiolus":
                expenses = numFlowers * 2.5;

                if (numFlowers < 80) {
                    expenses *= 1.20;
                }
                break;
        }
        if (budget >= expenses) {
            System.out.printf("Hey, you have a great garden with %d " +
                    "%s and %.2f leva left.", numFlowers, flowers, budget - expenses);
        } else {
            System.out.printf("Not enough money, you need %.2f leva more.", expenses - budget);
        }
    }
}
